test/unit/webauthn_verification_test.rb

```diff
@@ -69,4 +69,63 @@ class WebauthnVerificationTest < ActiveSupport::TestCase
       assert_equal @generated_time + 2.minutes, @webauthn_verification.otp_expires_at
     end
   end
+
+  context "#verify_otp" do
+    setup do
+      @user = create(:user)
+      travel_to Time.utc(2023, 1, 1, 0, 1, 0)
+      freeze_time
+    end
+
+    context "when otp is correct and not expired" do
+      setup do
+        @verification = create(:webauthn_verification, user: @user)
+      end
+
+      should "return true" do
+        assert @verification.verify_otp(@verification.otp)
+      end
+
+      should "expire otp" do
+        @verification.verify_otp(@verification.otp)
+        assert_equal Time.utc(2023, 1, 1, 0, 0, 59), @verification.otp_expires_at
+      end
+    end
+
+    context "when otp is incorrect" do
+      setup do
+        @expires_at = 2.minutes.from_now
```

Comment:
I think we can benefit from being more explicit with the time here.

What if we saved the "current time" in the test's general context to a instance variable (referring to Line 76).
```ruby
  context "#verify_otp" do
    setup do
      @user = create(:user)
      @current_time = Time.utc(2023, 1, 1, 0, 1, 0)

      travel_to @current_time
      # ... etc.
```

That way we can be explicit and indicate the relationship of the time in the setup of this test (and other ones, see comment below (Line 113)
```ruby
    context "when otp is incorrect" do
      setup do
        @expires_at = @current_time + 2.minutes
```

Reply:
That would make things more clear 👍  updated! (Along with the other comment)